### ISSUE TITLE
unexpected behaviour of whitelist on filesystem different from the one of $HOME

### ISSUE THREAD

**Issue by wcat** [netblue30/firejail#278]
I think this is not the expected behaviour (/media/data is the mount point of a filesystem that I mount on boot):

`touch /media/data/some-file /media/data/foo/some-file`
`firejail --blacklist=/media/data find / -name some-file 2> /dev/null`
no file is found
`firejail ---blacklist=/media/data --whitelist=/media/data/some-dir find / -name some-file 2> /dev/null`
finds: /run/firejail/mnt/orig-media/data/some-file and /run/firejail/mnt/orig-media/data/foo/some-file

Basically if I whitelist something on /media/data then everything is accessible.

firejail version: 0.9.36
os: Fedora 23 x86_64

**Comment by manevich** [comment]
Unable to reproduce on Debian Jessie with stock kernel, both with 0.9.36 and https://github.com/netblue30/firejail/commit/19afadb5e95cb6c2ede60e80b81f662beb6eb899.

Also, you are misusing `--whitelist` option, it's not opposite to `--blacklist`, with `firejail --whitelist=/media/data/some-dir` you should get `/media/data/` containing only `some-dir` directory.

**Comment by wcat** [comment]
ok by using only --whitelist it works as expected (only /media/data/some-dir  is visible). 

Blacklisting a directory and then whitelisting a directory (or file) inside it leads to different behaviors on my system:
- on $HOME everything is blacklisted and I can't see any file.
- on some other mount point (I've tried on /media/data and /data) whitelisting any file (or directory) completely remove the effects of the blacklisting. If possible I think that firejail in this situation should at least fail with some error message instead of whitelisting everything. Even if this is a misuse of firejail it could leads to a security problem if someone do this by mistake and didn't notice what have just happened.

**Comment by netblue30** [comment]
There is a more general bug here: https://github.com/netblue30/firejail/issues/276

Let me fix 276, and will take it from there. Thanks for the bug.

**Comment by netblue30** [comment]
267 fixed, give it another try.

**Comment by wcat** [comment]
yes it doesn't happen no more on 0.9.38.

Thanks
